### PR TITLE
Allow empty server context in shiny prerendered

### PR DESCRIPTION
**Pull request by cderv**
This closes https://github.com/rstudio/rmarkdown/issues/2272 

After looking into other more complex solution, I think this is the easiest and the more targeted to the issue. We don't change **knitr** behavior but only empty chunk with `context = "server"` by defaulting to a empty server function in that case.

Example provided is 
````markdown
---
title: "Hello Prerendered Shiny"
output: html_document
runtime: shinyrmd
---

```{r, echo=FALSE}
sliderInput("bins", "Number of bins:", min = 1, max = 50, value = 30)
```

```{r, context="server"}
```
````

With current **rmarkdown** version, it will lead to an error because no value is passed to context and no `server.R` file is provided. 
As explained in https://github.com/rstudio/rmarkdown/issues/2272#issuecomment-1034009694, this is because **knitr** won't call evaluate hook on an empty code chunk, and so the shiny prerendered magic does not apply. 

This PR insert a comment in the code chunk if an empty server is provided so that knitr trigger the evaluate hook and the context is registered in the HTML.  `shiny_prerendered_app()` generate this server function, 
````r
  server <- function(input, output, session) {
    eval(xfun::parse_only(.server_context))
  }
````
which will result in an empty server function for the app if `.server_context` is only a comment. 

Only possible breaking change I am seing is that previously, if you had a empty server context but a `server.R` file, the app would render. With this change, the `server.R` file won't be used. I don't think this situation would happen so I think it is fine to change. 

Do you think that solves your issue in the first place ? Do you see any side effect I may have not think of ?

### CODE REVIEW

File: R/util.R [hunk 1]
```diff
@@ -112,6 +112,15 @@ is_null_or_string <- function(text) {
   is.null(text) || (is.character(text) && (length(text) == 1))
 }
 
+# TODO: Export knitr:::is_blank() in xfun, and use here
```
Comment (yihui) [inline R/util.R:115]: Done: https://github.com/yihui/xfun/commit/6f4eddd0d74974a12bbc431053849495ac6bd1bc

**Review by yihui — APPROVED** [review]
This seems to be a reasonable workaround to me. Thanks!

File: R/util.R (hunk 1 above)
Comment (cderv) [reply R/util.R:115]: Thanks ! 

Should we use it now here ? Or usually, you go through this kind of TODO after a new **xfun** is released ?

Comment (yihui) [reply R/util.R:115]: No, not now, but after the next **xfun** release.

Yes, I search for "TODO" in each package before I start thinking about making a CRAN release of that package.

Comment (cderv) [reply R/util.R:115]: Perfect. I'll try to remember that ! 👍

Comment (yihui) [reply R/util.R:115]: Usually a TODO comment is not urgent to me, and can be done at any time in the future (not necessarily before the next release) if I remember.

**Comment by cderv** [comment]
@jcheng5 I'll merge this. I think it is ok and won't bother your any longer for a review. Please feel free to react if otherwise. Thanks !

**Pull request merged**